apps/multiplatform/common/src/commonMain/kotlin/chat/simplex/common/views/contacts/ContactsView.kt

```diff
@@ -0,0 +1,385 @@
+    val chat2Type = getContactType(chat2)
+
+    when {
+        chat1Type.ordinal < chat2Type.ordinal -> -1
+        chat1Type.ordinal > chat2Type.ordinal -> 1
+
+        else -> chat2.chatInfo.chatTs.compareTo(chat1.chatInfo.chatTs)
+    }
+}
+
+@Composable
+private fun ModalData.DeletedContactsView(rh: RemoteHostInfo?, close: () -> Unit) {
+    ModalView(
+        close = close
+    ) {
+        Column(
+            Modifier.fillMaxSize(),
+        ) {
+            Box(contentAlignment = Alignment.Center) {
+                val bottomPadding = DEFAULT_PADDING
+                AppBarTitle(
+                    stringResource(MR.strings.chat_deleted),
+                    hostDevice(rh?.remoteHostId),
+                    bottomPadding = bottomPadding
+                )
+            }
+
+            ContactsLayout(
+                contactActions = {},
+                contactTypes = listOf(ContactType.REMOVED),
+            )
+        }
+    }
+}
+
+@Composable
+private fun ContactActionsSection(contactActions: @Composable () -> Unit, rh: RemoteHostInfo?) {
+    contactActions()
+    Spacer(Modifier.height(DEFAULT_PADDING))
+
+    val archived = remember { contactChats(chatModel.chats, listOf(ContactType.REMOVED)) }
+
+    if (archived.isNotEmpty()) {
+        SectionView {
+            SectionItemView(
+                click = {
+                    ModalManager.start.showCustomModal { close -> DeletedContactsView(
+                        rh = rh,
+                        close = close)
+                    }
+                }
+            ) {
+                Icon(
+                    painterResource(MR.images.ic_folder_open),
+                    contentDescription = stringResource(MR.strings.chat_deleted),
+                    tint = MaterialTheme.colors.secondary,
+                )
+                TextIconSpaced(extraPadding = true)
+                Text(text = stringResource(MR.strings.chat_deleted), color = MaterialTheme.colors.onBackground)
+            }
+        }
+    }
+}
+
+@Composable
+private fun ContactsLayout(
+    contactActions: @Composable () -> Unit,
+    contactTypes: List<ContactType>,
+    contactListTitle: String? = null) {
+
+    SectionView {
+        ContactsList(
+            contactTypes = contactTypes,
+            contactActions = contactActions,
+            contactListTitle = contactListTitle
+        )
+    }
+}
+
+@Composable
+fun ContactsView(
+    contactActions: @Composable () -> Unit,
+    rh: RemoteHostInfo?
+) {
+    ContactsLayout(
+        contactActions = { ContactActionsSection(contactActions, rh) },
+        contactTypes = listOf(ContactType.CARD, ContactType.RECENT, ContactType.REQUEST),
+        contactListTitle = stringResource(MR.strings.contact_list_header_title).uppercase()
+    )
+}
+
+@Composable
+private fun ContactsSearchBar(listState: LazyListState, searchText: MutableState<TextFieldValue>, focused: Boolean, onFocusChanged: (hasFocus: Boolean) -> Unit) {
```

Comment:
since now it's not simply contacts view, but New chat view, should say "Search or paste SimpleX link" and correctly react to pasting it - offer to connect / open known chats, etc. - basically copy look and behavior of chat list search